### PR TITLE
update `catalogView.dtml` to changed behavior for "empty searches"

### LINKED ISSUE #102: "dtml/catalogView.dtml" needs update to changed behavior for "empty" searches
This originally has been #101 .

`searchResults` has recently been changed regarding "empty" searches: formerly, an "empty" search returned all catalogued objects, now, it returns 0 results.

The above mentioned dtml file must be updated accordingly. Otherwise, the `Catalog` tab reports an empty catalog whenever the catalog lacks an index named `path`.

### PR DESCRIPTION
**Pull request by d-maurer**
This fixes #102 (and #101).

It also interprets a missing or empty `filterpath` as "no restriction" (instead of `"/"`). This allows a potentially available `path` index to index generalized paths (not necessary starting with `"/"`).

### CODE REVIEW

File: src/Products/ZCatalog/dtml/catalogView.dtml [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 <main class="container-fluid">
 
-    <dtml-let filterpath="REQUEST.get('filterpath', '/')"
-              results="searchResults(path=filterpath)">
+    <dtml-let filterpath="REQUEST.get('filterpath', '')"
+              results="searchResults(path=filterpath) if filterpath else list(getAllBrains())">
```
Comment (ale-rt) [inline src/Products/ZCatalog/dtml/catalogView.dtml:7]: Isn't this a performance issue?

**Comment by d-maurer** [comment]
Alessandro Pisa wrote at 2020-6-18 08:49 -0700:
>@ale-rt commented on this pull request.
>
>
>
>> @@ -3,8 +3,8 @@
>
> <main class="container-fluid">
>
>-    <dtml-let filterpath="REQUEST.get('filterpath', '/')"
>-              results="searchResults(path=filterpath)">
>+    <dtml-let filterpath="REQUEST.get('filterpath', '')"
>+              results="searchResults(path=filterpath) if filterpath else list(getAllBrains())">
>
>Isn't this a performance issue?

I have defined a new method `searchAll`.
It is similar to `getAllBrains` but returns a data type
like the other searches (`LazyMap`) instead of an interator.
I have replaced the expensive `list(getAllBrains())` with `searchAll()`.

File: src/Products/ZCatalog/tests/test_zcatalog.py [hunk 2]
```diff
@@ -307,6 +314,16 @@ def testSearchNot(self):
...
     # _getProgressThreshold
 
+    def test_catalogView(self):
+        catalog = makerequest(self._catalog)
+        # hack `getPhysicalPath` to avoid problem with the catalog plan
+        catalog.getPhysicalPath = None
+        # provide `ZopeVersion`
+        catalog.ZopeVersion = lambda *arg, **kw: 4
+        vr = catalog.manage_catalogView()
+        self.assertTrue("There are no objects in the Catalog." not in vr,
```
Comment (icemac) [inline src/Products/ZCatalog/tests/test_zcatalog.py:324]:
```suggestion
        self.assertNotIn("There are no objects in the Catalog.", vr,
```

Maybe an assertion what's actually in `vr` would be helpful as asserting something is not there is a relatively weak assertion. There are so many reasons something might not be there.

Comment (d-maurer) [reply src/Products/ZCatalog/tests/test_zcatalog.py:324]: The test directly verifies the faulty behavior introduced by changing the way empty searches are handled ("all objects" -> "no objects"). It caused the message `There are no objects in the Catalog` even though this was false. The test checks that this faulty behavior is gone. Testing that the template actually produces the desired output requires much more machinery. I checked this aspect visually.

**Pull request merged**